### PR TITLE
feat: Release helm charts on GitHub pages

### LINKED ISSUE #4834: Use helm-chart repository to publish Keptn helm-charts
## Action Item

Currently, we store our helm charts inside a Google Storage bucket: https://storage.googleapis.com/keptn-installer. Other open-source projects (e.g., [Grafana](https://github.com/grafana/helm-charts)) leverage github and its [gh-pages](https://pages.github.com/) feature.

Keptn should also provide a `helm-charts` repository. Furthermore, it should also provide `helm-charts-dev` that contains the nightly build of Keptn from the `/master` branch. 

Additionally, the Keptn CLI shall also be hosted on gh-pages if possible.
:exclamation: Note that the CLI itself must also be adjusted.

**Please note:** Duplicate to: https://github.com/keptn/keptn/issues/2532, which already contains an ongoing discussion.

References:
https://helm.sh/docs/topics/chart_repository/#github-pages-example
https://helm.sh/docs/howto/chart_releaser_action/

## DoD 

- [x] Update the pipeline to release the helm charts to the `helm-charts` github repository
- [x] Update the pipeline to publish the builds form `/master` into the `helm-charts-dev` github repository
- [x] Update README.md with the command to install the latest dev version

### PR DESCRIPTION
**Pull request by mowies**
## This PR

- changes the release process to publish keptn helm charts to github pages
- deletes lots of unused shell scripts
- calculates the actual next version tag for dev builds instead of just using the next patch version every time
- bumps all pipelines to use Node 16 instead of 14 (no impact on built docker images based on node)

### Related Issues

Fixes #4834 
PR to update the documentation: https://github.com/keptn/keptn.github.io/pull/1005

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -357,6 +355,52 @@ jobs:
...
+        KEPTN_BOT_NAME: "Keptn Bot"
+        KEPTN_BOT_EMAIL: "keptn-bot <86361500+keptn-bot@users.noreply.github.com>"
+      run: |
+        git config user.name "$KEPTN_BOT_NAME"
+        git config user.email "$KEPTN_BOT_EMAIL"
+
+    - name: Install Helm
+      uses: azure/setup-helm@v1
+      with:
+        version: v3.7.2
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/CI.yml:384]: do we need a specific helm version? Can we make sure that the [build-helm-charts](https://github.com/keptn/keptn/blob/master/.github/workflows/build-helm-charts.yml) workflow uses the same helm version?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -184,9 +176,15 @@ jobs:
...
-          echo "GIT_LAST_TAG=${GIT_LAST_TAG}, GIT_NEXT_TAG=${GIT_NEXT_TAG}"
+          npx standard-version@^9.3.1 \
+            --skip.commit \
+            --skip.tag \
+            --skip.changelog
+
+          GIT_NEXT_TAG=$(cat VERSION.txt)
+          git checkout HEAD -- VERSION.txt
+          
+          echo "GIT_NEXT_TAG=${GIT_NEXT_TAG}"
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/CI.yml:187]: those changes should also make it into https://github.com/keptn/gh-automation/blob/main/.github/workflows/prepare-ci.yml

File: .github/workflows/pre-release.yml [hunk 3]
```diff
@@ -363,35 +374,38 @@ jobs:
...
-          export OS_TYPE="linux"
-          mkdir ~/downloads
-          ./test/utils/download_and_install_gcloud.sh
-          echo ${GCLOUD_SERVICE_KEY} | base64 --decode > ~/gcloud-service-key.json
-          gcloud auth activate-service-account --key-file ~/gcloud-service-key.json
-          gh-actions-scripts/upload_helm_chart_to_google_cloud.sh dist/keptn-installer/
+          mv "$ARTIFACTS" ./helm-charts/packages
+          cd ./helm-charts
+          helm repo index ./ --url https://charts.keptn.sh/ --merge ./index.yaml
+          git add --all
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/pre-release.yml:406]: This is mega dangerous. If for whatever reason there is a logfile with secrets or alike in this directory, we will leak it!
We should make sure that we add files by extension here.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -359,35 +370,38 @@ jobs:
...
-          export OS_TYPE="linux"
-          mkdir ~/downloads
-          ./test/utils/download_and_install_gcloud.sh
-          echo ${GCLOUD_SERVICE_KEY} | base64 --decode > ~/gcloud-service-key.json
-          gcloud auth activate-service-account --key-file ~/gcloud-service-key.json
-          gh-actions-scripts/upload_helm_chart_to_google_cloud.sh dist/keptn-installer/
+          mv "$ARTIFACTS" ./helm-charts/packages
+          cd ./helm-charts
+          helm repo index ./ --url https://charts.keptn.sh/ --merge ./index.yaml
+          git add --all
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/release.yml:402]: This is mega dangerous. If for whatever reason there is a logfile with secrets or alike in this directory, we will leak it!
We should make sure that we add files by extension here.

File: releasenotes/releasenotes_V0.8.5.md [hunk 5]
```diff
@@ -14,12 +14,12 @@ Installing and/or upgrading to this release should be done using the `helm upgra
 
 **Installation** (using `--install`)
 ```console
-helm upgrade keptn keptn --install -n keptn --create-namespace --wait --version=0.8.5 --repo=https://storage.googleapis.com/keptn-installer --set=control-plane.bridge.lookAndFeelUrl=https://example.com/bridge-look-and-feel.zip
+helm upgrade keptn keptn --install -n keptn --create-namespace --wait --version=0.8.5 --repo=https://charts.keptn.sh --set=control-plane.bridge.lookAndFeelUrl=https://example.com/bridge-look-and-feel.zip
```
Comment (christian-kreuzberger-dtx) [inline releasenotes/releasenotes_V0.8.5.md:17]: was this change intended?

File: releasenotes/releasenotes_V0.8.6.md [hunk 6]
```diff
@@ -15,12 +15,12 @@ Installing and/or upgrading to this release should be done using the `helm upgra
 
 **Installation** (using `--install`)
 ```console
-helm upgrade keptn keptn --install -n keptn --create-namespace --wait --version=0.8.6 --repo=https://storage.googleapis.com/keptn-installer
+helm upgrade keptn keptn --install -n keptn --create-namespace --wait --version=0.8.6 --repo=https://charts.keptn.sh
```
Comment (christian-kreuzberger-dtx) [inline releasenotes/releasenotes_V0.8.6.md:18]: was this change intended?

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
I left some comments

File: .github/workflows/CI.yml (hunk 2 above)
Comment (mowies) [reply .github/workflows/CI.yml:187]: I am not so sure, they only work with an annotated changelog according to conventional changelog rules. I don't know if all integration use that

File: releasenotes/releasenotes_V0.8.6.md (hunk 6 above)
Comment (mowies) [reply releasenotes/releasenotes_V0.8.6.md:18]: yes, the historical charts are also online in the new charts repo

File: releasenotes/releasenotes_V0.8.5.md (hunk 5 above)
Comment (mowies) [reply releasenotes/releasenotes_V0.8.5.md:17]: yes, the historical charts are also online in the new charts repo

File: .github/workflows/pre-release.yml (hunk 3 above)
Comment (mowies) [reply .github/workflows/pre-release.yml:406]: good catch, i will change it to the following:
```
git add *.tgz
git add index.yaml
```

File: .github/workflows/release.yml (hunk 4 above)
Comment (mowies) [reply .github/workflows/release.yml:402]: good catch, i will change it to the following:
```
git add *.tgz
git add index.yaml
```

File: .github/workflows/CI.yml (hunk 1 above)
Comment (mowies) [reply .github/workflows/CI.yml:384]: not really. if it's left out, the latest stable version will be used

Comment (mowies) [reply .github/workflows/CI.yml:384]: actually, helm is included on ubuntu GH actions runner per default anyways so i will remove the setup step

File: .github/workflows/CI.yml (hunk 2 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/CI.yml:187]: So far only a couple integrations use the prepare-ci step from keptn/gh-automation, and those already use `standard-version` package ;)

Comment (mowies) [reply .github/workflows/CI.yml:187]: cool, then i will add it there as well

Comment (mowies) [reply .github/workflows/CI.yml:187]: this still needs the post-changelog script that i implemented to update the version.txt file with the new version string and specific configuration and such. I still think it's too specialized for keptn/keptn specifically

Comment (christian-kreuzberger-dtx) [reply .github/workflows/CI.yml:187]: ok

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -357,6 +355,52 @@ jobs:
...
+        path: dist/keptn-installer
+
+    - name: Upload helm charts to dev repo
+      env:
+        RELEASE_TAG: ${{ needs.prepare_ci_run.outputs.VERSION }}
+      run: |
+        mv ./dist/keptn-installer/*.tgz ./helm-charts-dev/packages
+        cd ./helm-charts-dev
+        helm repo index ./ --url https://charts-dev.keptn.sh/ --merge ./index.yaml
+        git add --all
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/CI.yml:400]: Should we do `git add *.tgz` here too?

Comment (mowies) [reply .github/workflows/CI.yml]: good catch! done :)

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM
